src/svg/qsvggenerator.cpp

```diff
@@ -1049,50 +1088,61 @@ void QSvgPaintEngine::drawEllipse(const QRectF &r)
     *d->stream << "\"/>" << endl;
 }
 
-void QSvgPaintEngine::drawPath(const QPainterPath &p)
+QString QSvgPaintEngine::createPath(const QPainterPath &p)
 {
-    Q_D(QSvgPaintEngine);
+    QTextStream path_stream;
+    QString path;
+
+    path_stream.setString(&path);
 
-    *d->stream << "<path vector-effect=\""
-               << (state->pen().isCosmetic() ? "non-scaling-stroke" : "none")
-               << "\" fill-rule=\""
-               << (p.fillRule() == Qt::OddEvenFill ? "evenodd" : "nonzero")
-               << "\" d=\"";
+    path_stream << "<path vector-effect=\""
+               << (state->pen().isCosmetic() ? "non-scaling-stroke" : "none") << "\" fill-rule=\""
+               << (p.fillRule() == Qt::OddEvenFill ? "evenodd" : "nonzero") << "\" d=\"";
```

Comment:
I find the old structure here much easier to read, but not a huge issue

Reply:
(i.e. splitting the stream over more lines)

Reply:
I think this was caused by my IDE auto-formatting, I have disabled it but it won't stop... :(